src/Cli/dotnet/ParseResultExtensions.cs

```diff
@@ -106,5 +107,22 @@ internal static string GetCommandLineRuntimeIdentifier(this ParseResult parseRes
                     parseResult.ValueForOption<string>(CommonOptions.ArchitectureOption().Aliases.First())) :
                 null;
         }
+
+        public static bool UsingRunCommandShorthandProjectOption(this ParseResult parseResult)
```

Comment:
No, I don't think there's a way to differentiate, without separating it into a separate option, which we don't want to do because then it wouldn't show up in the help.

Reply:
Opened https://github.com/dotnet/sdk/issues/19514 to track this